### PR TITLE
test: add test for network teardown on sandbox error

### PR DESCRIPTION
**Pull request by dcbw**
Signed-off-by: Dan Williams <dcbw@redhat.com>

### CODE REVIEW

**Comment by dcbw** [comment]
(note, don't merge this yet; I failed to get testing running locally so I'm abusing travis)

**Comment by runcom** [comment]
`make integration TESTFLAGS=network.bats` should run tests in containers just fine (otherwise we have some issues)

**Comment by dcbw** [comment]
ah, didn't realize bats did that for us, and forgot that I'd done that before... was trying to run the integration tests with root manually

**Comment by dcbw** [comment]
@runcom have you seen this before?

```
touch "/go/.gopathok"
go build -ldflags '-X main.gitCommit="8b3a688c21242a393eef7dd531295bf0443fc064" -X main.buildInfo=1505337859 -X main.version=1.0.0-rc2-dev -X main.kpodVersion=0.1' -tags "selinux seccomp  " -o test/copyimg/copyimg github.com/kubernetes-incubator/cri-o/test/copyimg
time="2017-09-13T21:24:44Z" level=error msg="error copying docker://redis to dir:.artifacts/redis-image: can not copy docker://redis:latest: manifest contains multiple images" 
The command '/bin/sh -c make test/copyimg/copyimg 	&& mkdir -p .artifacts/redis-image 	&& ./test/copyimg/copyimg --import-from=docker://redis --export-to=dir:.artifacts/redis-image --signature-policy ./test/policy.json' returned a non-zero code: 1
Makefile:110: recipe for target 'crioimage' failed
make: *** [crioimage] Error 1
```

File: test/network.bats [hunk 1]
```diff
@@ -118,3 +118,17 @@ load helpers
 
 	stop_crio
 }
+
+@test "Clean up network if pod sandbox fails" {
+	rm -f /var/lib/cni/networks/crionet_test_args/*
+	start_crio "" "" "" "prepare_plugin_test_args_network_conf"
+	# kill conmon to ensure pod fails to start after networking is set up
+	kill -9 $(ps -C conmon -o pid=)
```
Comment (mrunalp) [inline test/network.bats]: There should be no conmon running without starting a pod or container.

**Comment by runcom** [comment]
You need to rebase to latest master, the fix for that went in half hour ago with #904

File: test/network.bats (hunk 1 above)
Comment (dcbw) [reply test/network.bats]: @mrunalp ah, ok.  what I'm trying to do is cause a failure of sandbox setup *after* network has been correctly set up.  When testing this locally I had an old conmon installed and that did the trick, any suggestions here?  The only operations after network are:
```
	err = g.SaveToFile(filepath.Join(podContainer.Dir, "config.json"), saveOptions)
	if err != nil {
		return nil, fmt.Errorf("failed to save template configuration for pod sandbox %s(%s): %v", sb.Name(), id, err)
	}
	if err = g.SaveToFile(filepath.Join(podContainer.RunDir, "config.json"), saveOptions); err != nil {
		return nil, fmt.Errorf("failed to write runtime configuration for pod sandbox %s(%s): %v", sb.Name(), id, err)
	}

	if err = s.runContainer(container, sb.CgroupParent()); err != nil {
		return nil, err
	}

	s.addInfraContainer(container)

	s.ContainerStateToDisk(container)
```

Comment (dcbw) [reply test/network.bats]: perhaps I could have a wrapper script for the network plugin that removes podContainer.Dir or something...

Comment (mrunalp) [reply test/network.bats]: @dcbw Yeah, something like that should work.

**Comment by dcbw** [comment]
@runcom @mrunalp ok, I now have a test that actually works...

File: test/network.bats [hunk 2]
```diff
@@ -118,3 +125,22 @@ load helpers
 
 	stop_crio
 }
+
+@test "Clean up network if pod sandbox fails" {
+	rm -f /var/lib/cni/networks/crionet_test_args/*
```
Comment (runcom) [inline test/network.bats]: Can we put this in the tear down?

**Comment by runcom** [comment]
LGTM (just a silly nit, thanks @dcbw)

File: test/network.bats (hunk 2 above)
Comment (TomSweeneyRedHat) [reply test/network.bats]: 👍

**Comment by dcbw** [comment]
@runcom @TomSweeneyRedHat updated for your comments, PTAL thanks!

File: test/network.bats [hunk 3]
```diff
@@ -2,6 +2,14 @@
 
 load helpers
 
+function teardown() {
+	cleanup_ctrs
```
Comment (TomSweeneyRedHat) [inline test/network.bats]: I think it's OK, but want to double check.  If you run the cleanup_ctrs or cleanup_pods after a test that didn't create one or the other, it doesn't create an error correct?  I think not, but I'm not 100% sure.

Comment (dcbw) [reply test/network.bats]: @TomSweeneyRedHat correct, it doesn't create an error.  A couple of the tests in network.bats do not create containers, and the tests pass even when cleanup_ctrs() is called.

Comment (TomSweeneyRedHat) [reply test/network.bats]: LGTM, thanks for the follow up @dcbw

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by rhatdan** [comment]
[retest]

**Comment by dcbw** [comment]
@runcom @mrunalp anything else I need to do for this one?

**Comment by mrunalp** [comment]
@dcbw Sorry for the delay. Our CI has been dead that's why PRs are taking longer to merge. Hopefully it should be back up soon.

**Comment by mrunalp** [comment]
@dcbw if you rebase the PR to master, we can run e2e tests now.

**Comment by dcbw** [comment]
@mrunalp rebased, tests pass now.

**Comment by mrunalp** [comment]
The fedora error is known. Will wait for e2e_rhel.

**Comment by rhatdan** [comment]
RHEL has 8 failures.

**Comment by mrunalp** [comment]
I have fixed the test issues in #969 so rebase should fix the e2e. Sorry about this. We moved to a new base ami which took longer than expected.

**Comment by dcbw** [comment]
Test flake timeout on image pull...

/test all

**Comment by mrunalp** [comment]
The tests look good (only known failures).

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**